### PR TITLE
Allow sitewide asset opt-out in production

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1211
Relates to https://github.com/department-of-veterans-affairs/content-build/pull/1358

This will allow the particular script and link tags to be excluded from production pages if an application has a truthy `useLocalStylesAndComponents` in its registry.

Right now the `pre-check-in` application is the only one that will be impacted by this change:

https://github.com/department-of-veterans-affairs/content-build/blob/69dbfa52df4c220c4827e2f35b583ad8c7973bc5/src/applications/registry.json#L1113-L1122

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

1. Do this
   - [ ] Validate that
2. Then
   - [ ] Validate that
3. Then validate Acceptance Criteria from issue

## Acceptance criteria

- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/includes/header.html [hunk 1]
```diff
@@ -47,7 +47,7 @@
       this global file which will define all web components. Instead, they will manually
       define a subset of web components which are used in their application
     {% endcomment %}
-    {% if buildtype == 'vagovprod' or !useLocalStylesAndComponents%}
+    {% if !useLocalStylesAndComponents%}
       <script nonce="**CSP_NONCE**" defer data-entry-name="web-components.js"></script>
     {% endif %}
```
Comment (acrollet) [inline src/site/includes/header.html:52]:
```suggestion
    {% unless useLocalStylesAndComponents %}
      <script nonce="**CSP_NONCE**" defer data-entry-name="web-components.js"></script>
    {% endunless %}
```

File: src/site/includes/header.html [hunk 2]
```diff
@@ -76,7 +76,7 @@
     Apps that opt out of the site-wide stylesheet must include a smaller subset of styles
     in their application bundle
   {% endcomment %}
-  {% if buildtype == 'vagovprod' or !useLocalStylesAndComponents %}
+  {% if !useLocalStylesAndComponents %}
     <link rel="stylesheet" data-entry-name="style.css">
   {% endif %}
```
Comment (acrollet) [inline src/site/includes/header.html:81]:
```suggestion
  {% unless useLocalStylesAndComponents %}
    <link rel="stylesheet" data-entry-name="style.css">
  {% endunless %}
```

**Review by acrollet — CHANGES_REQUESTED** [review]
The logical NOT operator does not exist in liquid, but these suggestions should work. (Discovered this while working on #1425)

**Review by longmd — APPROVED** [review]
LGTM 🚀

**Review by jtmst — APPROVED** [review]

**Review by acrollet — APPROVED** [review]
We reviewed this today and it's good to go from our perspective - only ask is that we can coordinate when this gets merged because we'd like to be able to validate that things function correctly in prod when it goes out. Thanks!

**Comment by Andrew565** [comment]
> only ask is that we can coordinate when this gets merged

If it's more convenient, I can re-assign this PR to you and you can handle the merge when you're ready?

**Pull request merged**